web_src/js/features/common-page.ts

```diff
@@ -91,3 +91,12 @@ export function checkAppUrl() {
   showGlobalErrorMessage(`Your ROOT_URL in app.ini is "${appUrl}", it's unlikely matching the site you are visiting.
 Mismatched ROOT_URL config causes wrong URL links for web UI/mail content/webhook notification/OAuth2 sign-in.`, 'warning');
 }
+
+export function checkAppUrlScheme() {
+  const curUrl = window.location.href;
+  // some users visit "http://domain" while appUrl is "https://domain", COOKIE_SECURE makes it impossible to sign in
+  if (curUrl.startsWith('http:') && appUrl.startsWith('https:')) {
+    showGlobalErrorMessage(`Your ROOT_URL in app.ini "${appUrl}" uses HTTPS, it doesn't match the HTTP site you are visiting.
+Mismatched ROOT_URL config would cause problems for sign-in/sign-up, etc.`, 'warning');
```

Comment:
```suggestion
    showGlobalErrorMessage(`This instance is configured to run under "${appUrl}". You are accessing it using "${curUrl}".
You will likely encounter problems when trying to sign in, register an account, or similar.`, 'warning');
```

Reply:
The only difference is HTTPS vs HTTP, so it doesn't seem useful to mention the full URL.

Reply:
And ROOT_URL should be explicitly mentioned to make site admin know where to go on. See another old notice above, it does help a lot of users, and reduces a lot of duplicate (non)bug reports in the issue tracker.

![image](https://github.com/user-attachments/assets/743a1f92-99bf-4a0a-bcb2-ac2ac43557d7)

Reply:
Well, if we want to keep it specific to that, what about
```suggestion
    showGlobalErrorMessage(`This instance is configured to run under "${appUrl}". You are trying to access it in unsafe way ('http' instead of 'https').
You will likely encounter problems when trying to sign in, register an account, or similar.`, 'warning');
```

Reply:
And one more thing, `This instance is configured to run under "${appUrl}` is not accurate now, because in most cases Gitea doesn't require need the ROOT_URL anymore, theoretically Gitea could run under any domain now (see my ROOT_URL/domain related PRs)

Reply:
And one more thing, `This instance is configured to run under "${appUrl}` is not accurate now, because in most cases Gitea doesn't require need the ROOT_URL anymore, theoretically Gitea could run under any domain now (see my ROOT_URL/domain related PRs)

Reply:
`You are trying to access it in unsafe way ('http' instead of 'https').` I do not think it is  good to be opinioned to tell users that they are "unsafe", maybe they are using it in a private network and tends to do so .....

The root problem is "https" in ROOT_URL makes "COOKIE_SECURE=true" and then http cookie stops working.

Reply:
I am not a native speaker, but I am pretty sure that we should:

1. Tell users that they are using https in ROOT_URL but they are accessing http, it might cause problems.
    * Site admin could change the ROOT_URL to use http if they really need when they see the warning.
2. Not mislead users to think that Gitea could only run under one ROOT_URL.
2. Not make judgments for users about "secure" or "insecure".

Reply:
```suggestion
    showGlobalErrorMessage(`This instance is configured to run under HTTPS (by ROOT_URL config), you are accessing by HTTP. Mismatched scheme might cause problems for sign-in/sign-up.`, 'warning');
```

Reply:
Sounds much better.

Reply:
Thank you for helping improving it!